### ISSUE TITLE
`mix phx.new` isn't working anymore..?

### ISSUE THREAD

**Issue by lucavenir** [phoenixframework/phoenix#5807]
### Environment

* Elixir version (elixir -v):
```
Erlang/OTP 26 [erts-14.2.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit:ns]
Elixir 1.16.2 (compiled with Erlang/OTP 26)
```
* Phoenix version (mix deps):
none added by hand, the following are added when running `mix new.phx`
```
* bandit 1.5.1 (Hex package) (mix)
  locked at 1.5.1 (bandit) a8200c92
  ok
* castore 1.0.7 (Hex package) (mix)
  locked at 1.0.7 (castore) da7785a4
  ok
* db_connection 2.6.0 (Hex package) (mix)
  locked at 2.6.0 (db_connection) c2f992d1
  ok
* decimal 2.1.1 (Hex package) (mix)
  locked at 2.1.1 (decimal) 53cfe5f4
  ok
* dns_cluster 0.1.3 (Hex package) (mix)
  locked at 0.1.3 (dns_cluster) 46cb7c4a
  ok
* ecto 3.11.2 (Hex package) (mix)
  locked at 3.11.2 (ecto) 3c38bca2
  ok
* ecto_sql 3.11.1 (Hex package) (mix)
  locked at 3.11.1 (ecto_sql) ce14063a
  ok
* esbuild 0.8.1 (Hex package) (mix)
  locked at 0.8.1 (esbuild) 25fc876a
  ok
* expo 0.5.2 (Hex package) (mix)
  locked at 0.5.2 (expo) 8c9bfa06
  ok
* file_system 1.0.0 (Hex package) (mix)
  locked at 1.0.0 (file_system) 6752092d
  ok
* finch 0.18.0 (Hex package) (mix)
  locked at 0.18.0 (finch) 69f5045b
  ok
* gettext 0.24.0 (Hex package) (mix)
  locked at 0.24.0 (gettext) bdf75cdf
  ok
* heroicons (https://github.com/tailwindlabs/heroicons.git - v2.1.1)
  locked at 88ab3a0 (tag: v2.1.1)
  ok
* hpax 0.1.2 (Hex package) (mix)
  locked at 0.1.2 (hpax) 2c87843d
  ok
* jason 1.4.1 (Hex package) (mix)
  locked at 1.4.1 (jason) fbb01ecd
  ok
* mime 2.0.5 (Hex package) (mix)
  locked at 2.0.5 (mime) da0d64a3
  ok
* mint 1.6.0 (Hex package) (mix)
  locked at 1.6.0 (mint) 3c5ae85d
  ok
* nimble_options 1.1.0 (Hex package) (mix)
  locked at 1.1.0 (nimble_options) 8bbbb394
  ok
* nimble_pool 1.1.0 (Hex package) (mix)
  locked at 1.1.0 (nimble_pool) af2e4e6b
  ok
* phoenix 1.7.12 (Hex package) (mix)
  locked at 1.7.12 (phoenix) d646192f
  ok
* phoenix_ecto 4.5.1 (Hex package) (mix)
  locked at 4.5.1 (phoenix_ecto) ebe43aa5
  ok
* phoenix_html 4.1.1 (Hex package) (mix)
  locked at 4.1.1 (phoenix_html) f2f2df5a
  ok
* phoenix_live_dashboard 0.8.3 (Hex package) (mix)
  locked at 0.8.3 (phoenix_live_dashboard) f9470a0a
  ok
* phoenix_live_reload 1.5.3 (Hex package) (mix)
  locked at 1.5.3 (phoenix_live_reload) b4ec9cd7
  ok
* phoenix_live_view 0.20.14 (Hex package) (mix)
  locked at 0.20.14 (phoenix_live_view) 82f6d006
  ok
* phoenix_pubsub 2.1.3 (Hex package) (mix)
  locked at 2.1.3 (phoenix_pubsub) bba06bc1
  ok
* phoenix_template 1.0.4 (Hex package) (mix)
  locked at 1.0.4 (phoenix_template) 2c0c81f0
  ok
* plug 1.15.3 (Hex package) (mix)
  locked at 1.15.3 (plug) cc4365a3
  ok
* plug_crypto 2.1.0 (Hex package) (mix)
  locked at 2.1.0 (plug_crypto) 131216a4
  ok
* postgrex 0.17.5 (Hex package) (mix)
  locked at 0.17.5 (postgrex) 50b8b11a
  ok
* swoosh 1.16.5 (Hex package) (mix)
  locked at 1.16.5 (swoosh) b2324cf6
  ok
* tailwind 0.2.2 (Hex package) (mix)
  locked at 0.2.2 (tailwind) ccfb5025
  ok
* telemetry 1.2.1 (Hex package) (rebar3)
  locked at 1.2.1 (telemetry) dad9ce9d
  ok
* telemetry_metrics 1.0.0 (Hex package) (mix)
  locked at 1.0.0 (telemetry_metrics) f23713b3
  ok
* telemetry_poller 1.1.0 (Hex package) (rebar3)
  locked at 1.1.0 (telemetry_poller) 9eb9d9cb
  ok
* thousand_island 1.3.5 (Hex package) (mix)
  locked at 1.3.5 (thousand_island) 2be69549
  ok
* websock 0.5.3 (Hex package) (mix)
  locked at 0.5.3 (websock) 6105453d
  ok
* websock_adapter 0.5.6 (Hex package) (mix)
  locked at 0.5.6 (websock_adapter) e04378d2
  ok
```
* Operating system:
```Ubuntu 22.04 LTS```

### Actual behavior
I tried and experimented with phoenix in the last month; so I wanted to expand via a first "more serious" project.
But now I'm not able to get `Up And Running` anymore 😢 Or at least not without encountering an error.

[Up And Running](https://hexdocs.pm/phoenix/up_and_running.html) says we should run these:
```
mix phx.new hello
cd hello
mix ecto.create
mix phx.server
```
To correctly start the server. Which works.

**But** whenever I hit `http://localhost:4000/`, the server reports this error
```
[info] GET /
[debug] Processing with HelloWeb.PageController.home/2
  Parameters: %{}
  Pipelines: [:browser]
[debug] Plug.Session could not verify incoming session cookie. This may happen when the session settings change or a stale cookie is sent.
[info] Sent 200 in 125ms
[error] GenServer #PID<0.681.0> terminating
** (Protocol.UndefinedError) protocol String.Chars not implemented for {Hello.Repo, :dynamic_repo} of type Tuple. This protocol is implemented for the following type(s): Atom, BitString, Date, DateTime, Decimal, Float, Integer, List, NaiveDateTime, Phoenix.LiveComponent.CID, Postgrex.Copy, Postgrex.Query, Time, URI, Version, Version.Requirement
    (elixir 1.16.2) lib/string/chars.ex:3: String.Chars.impl_for!/1
    (elixir 1.16.2) lib/string/chars.ex:22: String.Chars.to_string/1
    (bandit 1.5.1) lib/bandit/http1/handler.ex:37: anonymous fn/1 in Bandit.HTTP1.Handler.clear_process_dict/0
    (elixir 1.16.2) lib/enum.ex:987: Enum."-each/2-lists^foreach/1-0-"/2
    (bandit 1.5.1) lib/bandit/http1/handler.ex:26: Bandit.HTTP1.Handler.maybe_keepalive/2
    (bandit 1.5.1) lib/bandit/delegating_handler.ex:18: Bandit.DelegatingHandler.handle_data/3
    (bandit 1.5.1) /home/venir/Documents/venir.dev/letmego/hello/deps/thousand_island/lib/thousand_island/handler.ex:411: Bandit.DelegatingHandler.handle_continue/2
    (stdlib 5.2.1) gen_server.erl:1085: :gen_server.try_handle_continue/3
    (stdlib 5.2.1) gen_server.erl:995: :gen_server.loop/7
    (stdlib 5.2.1) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
Last message: {:continue, :handle_connection}
State: {%ThousandIsland.Socket{socket: #Port<0.23>, transport_module: ThousandIsland.Transports.TCP, read_timeout: 60000, silent_terminate_on_error: false, span: %ThousandIsland.Telemetry{span_name: :connection, telemetry_span_context: #Reference<0.2608432043.774111238.62414>, start_time: -576460741391933468, start_metadata: %{remote_port: 48622, remote_address: {127, 0, 0, 1}, telemetry_span_context: #Reference<0.2608432043.774111238.62414>, parent_telemetry_span_context: #Reference<0.2608432043.774111237.62749>}}}, %{opts: %{http: [], websocket: [], http_1: [], http_2: []}, plug: {Phoenix.Endpoint.SyncCodeReloadPlug, {HelloWeb.Endpoint, []}}, handler_module: Bandit.InitialHandler, http_1_enabled: true, http_2_enabled: true}}
```

### Expected behavior
No `[error]` should ever appear on a `GET /` request... Especially after following the simplest up-and-running guide on the web 😄

The error points towards `ThousandIsland` and `Bandit`, so I'm not sure how to solve this problem.

**Comment by chrismccord** [comment]
This looks like an issue that was introduced with Bandit 1.5.1. In the meantime, you can set bandit was `{:bandit, "1.5.0"}`. Thanks for the report! We'll coordinate on the bandit side to get this fixed up.

**Comment by lucavenir** [comment]
Thank you - I did not realize there was a `bandit` upgrade. Could I ask how I could have noticed this one by myself and prevented the "spam" in here?

**Comment by lucavenir** [comment]
Hold on, my `mix.exs` file contains the following dependencies:
```exs
defmodule Hello.MixProject do
  use Mix.Project

  def project do
    [
      app: :hello,
      version: "0.1.0",
      elixir: "~> 1.14",
      elixirc_paths: elixirc_paths(Mix.env()),
      start_permanent: Mix.env() == :prod,
      aliases: aliases(),
      deps: deps()
    ]
  end

  # Configuration for the OTP application.
  #
  # Type `mix help compile.app` for more information.
  def application do
    [
      mod: {Hello.Application, []},
      extra_applications: [:logger, :runtime_tools]
    ]
  end

  # Specifies which paths to compile per environment.
  defp elixirc_paths(:test), do: ["lib", "test/support"]
  defp elixirc_paths(_), do: ["lib"]

  # Specifies your project dependencies.
  #
  # Type `mix help deps` for examples and options.
  defp deps do
    [
      {:phoenix, "~> 1.7.12"},
      {:phoenix_ecto, "~> 4.4"},
      {:ecto_sql, "~> 3.10"},
      {:postgrex, ">= 0.0.0"},
      {:phoenix_html, "~> 4.0"},
      {:phoenix_live_reload, "~> 1.2", only: :dev},
      {:phoenix_live_view, "~> 0.20.2"},
      {:floki, ">= 0.30.0", only: :test},
      {:phoenix_live_dashboard, "~> 0.8.3"},
      {:esbuild, "~> 0.8", runtime: Mix.env() == :dev},
      {:tailwind, "~> 0.2", runtime: Mix.env() == :dev},
      {:heroicons,
       github: "tailwindlabs/heroicons",
       tag: "v2.1.1",
       sparse: "optimized",
       app: false,
       compile: false,
       depth: 1},
      {:swoosh, "~> 1.5"},
      {:finch, "~> 0.13"},
      {:telemetry_metrics, "~> 1.0"},
      {:telemetry_poller, "~> 1.0"},
      {:gettext, "~> 0.20"},
      {:jason, "~> 1.2"},
      {:dns_cluster, "~> 0.1.1"},
      {:bandit, "~> 1.2"}  # !!! 1.2, not 1.5.1
    ]
  end

  # Aliases are shortcuts or tasks specific to the current project.
  # For example, to install project dependencies and perform other setup tasks, run:
  #
  #     $ mix setup
  #
  # See the documentation for `Mix` for more info on aliases.
  defp aliases do
    [
      setup: ["deps.get", "ecto.setup", "assets.setup", "assets.build"],
      "ecto.setup": ["ecto.create", "ecto.migrate", "run priv/repo/seeds.exs"],
      "ecto.reset": ["ecto.drop", "ecto.setup"],
      test: ["ecto.create --quiet", "ecto.migrate --quiet", "test"],
      "assets.setup": ["tailwind.install --if-missing", "esbuild.install --if-missing"],
      "assets.build": ["tailwind hello", "esbuild hello"],
      "assets.deploy": [
        "tailwind hello --minify",
        "esbuild hello --minify",
        "phx.digest"
      ]
    ]
  end
end
```

what should I put instead of `1.2`, then?

**Comment by chrismccord** [comment]
We generate a mix.exs reference for you like       `{:bandit, "~> 1.5"}`, so it will pick up the latest compatible release. In this case, it picked up a bad release. So nothing you could look out for or stay on top of. Just a simple bug which will be fixed up shortly :)

**Comment by chrismccord** [comment]
You specifically want to lock it at `{:bandit, "1.5.0"}`

**Comment by lucavenir** [comment]
Thank you a ton for your patience 🙏🏼
Unluckily, I cannot help further.
Have a great day!

**Comment by mtrudel** [comment]
Bandit 1.5.2 released with a fix for this

**Comment by mtrudel** [comment]
@lucavenir I'd just released 1.5.0 a few hours ago so you were among the few unlucky first users. Sorry for the trouble!

**Comment by lucavenir** [comment]
You guys are incredible!

> Sorry for the trouble!

Don't you dare! So **thank you** for answering this quickly!
Have a great day!
